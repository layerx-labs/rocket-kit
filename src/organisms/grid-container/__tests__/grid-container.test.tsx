import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { GridContainer, GridRow, GridCol } from '../../..';

describe('GridContainer', () => {
  it('renders', () => {
    const { asFragment } = render(
      <GridContainer>
        <GridRow>
          <GridCol>
            Col 1ldhf lasdjfh lkjasdhflkasd fklahds fkljahs dklfjh asdklfh asl
          </GridCol>
          <GridCol>Col 2</GridCol>
          <GridCol size={2}>Col 3</GridCol>
        </GridRow>
        <GridRow>
          <GridCol>Col 1</GridCol>
          <GridCol>Col 2</GridCol>
        </GridRow>
      </GridContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
