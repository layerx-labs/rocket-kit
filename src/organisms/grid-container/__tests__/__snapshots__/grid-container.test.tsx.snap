// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridContainer renders 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (min-width:900px) {
  .c1 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:900px) {
  .c2 {
    -webkit-flex: 1;
    -ms-flex: 1;
    flex: 1;
  }
}

@media (min-width:900px) {
  .c3 {
    -webkit-flex: 2;
    -ms-flex: 2;
    flex: 2;
  }
}

<div
    class="c0 grid-container"
  >
    <div
      class="c1 grid-row"
    >
      <div
        class="c2 grid-col"
        size="1"
      >
        Col 1
      </div>
      <div
        class="c2 grid-col"
        size="1"
      >
        Col 2
      </div>
      <div
        class="c3 grid-col"
        size="2"
      >
        Col 3
      </div>
    </div>
    <div
      class="c1 grid-row"
    >
      <div
        class="c2 grid-col"
        size="1"
      >
        Col 1
      </div>
      <div
        class="c2 grid-col"
        size="1"
      >
        Col 2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
